### PR TITLE
[feature]: weets api integration

### PR DESCRIPTION
**Pull request by yovanny-wolox**
## Summary

Weets service api integration

## Known Issues

None

## Trello Card

https://trello.com/c/pb87KDg2/32-servicios-promises-witter

## Screenshot
Screenshot from 2021-07-09 19-09-54

### CODE REVIEW

File: app/errors.js [hunk 1]
```diff
@@ -8,3 +8,6 @@ exports.databaseError = message => internalError(message, exports.DATABASE_ERROR
 
 exports.DEFAULT_ERROR = 'default_error';
 exports.defaultError = message => internalError(message, exports.DEFAULT_ERROR);
+
+exports.BAD_GATEWAY = 'bad_gateway';
```
Comment (marioportela1993) [inline app/errors.js]: Here It is better to rename as `exports.BAD_GATEWAY_ERROR` in order to specify that variable as an error :)

File: app/middlewares/errors.js [hunk 2]
```diff
@@ -4,8 +4,9 @@ const logger = require('../logger');
 const DEFAULT_STATUS_CODE = 500;
 
 const statusCodes = {
-  [errors.DATABASE_ERROR]: 503,
-  [errors.DEFAULT_ERROR]: 500
+  [errors.DEFAULT_ERROR]: 500,
+  [errors.BAD_GATEWAY]: 502,
```
Comment (marioportela1993) [inline app/middlewares/errors.js]: Same here

File: app/errors.js [hunk 3]
```diff
@@ -8,3 +8,6 @@ exports.databaseError = message => internalError(message, exports.DATABASE_ERROR
...
+exports.badGateway = message => internalError(message, exports.BAD_GATEWAY);
```
Comment (marioportela1993) [inline app/errors.js]: Same here

File: app/services/weets.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const axios = require('axios');
```
Comment (marioportela1993) [inline app/services/weets.js]: You can create a function that wraps the axios package and call it from anywhere in the app

**Review by marioportela1993 — CHANGES_REQUESTED** [review]
Excellent job, check the comments and implement them

File: app/errors.js (hunk 1 above)
Comment (yovanny-wolox) [reply app/errors.js]: Done!

File: app/middlewares/errors.js (hunk 2 above)
Comment (yovanny-wolox) [reply app/middlewares/errors.js]: Done!

File: app/errors.js (hunk 3 above)
Comment (yovanny-wolox) [reply app/errors.js]: Done!

File: app/services/weets.js (hunk 4 above)
Comment (yovanny-wolox) [reply app/services/weets.js]: Done! thanks by the feedback

File: app/helpers/axios.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const axios = require('axios');
+
+const logger = require('../logger');
+const { badGatewayError } = require('../errors');
+
+exports.getWeetRandom = (method, url, data = '') =>
```
Comment (marioportela1993) [inline app/helpers/axios.js]: We can rename this method to be more generic and can be used for requesting whatever external service. Maybe something like `request`

File: app/helpers/axios.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  axios({ method, url, data })
+    .then(response => response.data.joke)
```
Comment (marioportela1993) [inline app/helpers/axios.js]: And here only return `response.data` in accordance with de above comment.

**Review by marioportela1993 — CHANGES_REQUESTED** [review]

File: app/helpers/axios.js (hunk 6 above)
Comment (yovanny-wolox) [reply app/helpers/axios.js]: it's was an oversight. Sorry!

File: app/helpers/axios.js (hunk 5 above)
Comment (yovanny-wolox) [reply app/helpers/axios.js]: I agree. Thanks

**Review by marioportela1993 — APPROVED** [review]

**Pull request merged**